### ISSUE TITLE
Taffy does not see my resources folder

### ISSUE THREAD

**Issue by d1b1** [atuttle/Taffy#234]
Hi I am trying to get taffy working on my windows (CF9) setup. I have been able to get the resources to show on a docker instance using (CF10). But when I start the code on the windows CF version It tells me that 'Taffy is running but you haven't defined any resources yet.' and tells me to create a `resources` folder where I already have one.

Any ideas? Is there a way to see where the taffy CFC is looking for resources?

**Comment by codybe** [comment]
Currently having the same problem as d1b1.  However the api example included has no trouble finding its resource folder.  File permissions look to be the same as the examples and I am extending taffy.core.resources.  

Has anyone figured a way to see were taffy is looking for the resources folder?

**Comment by atuttle** [comment]
You're not using symlinks anywhere in your path are you?

**Comment by codybe** [comment]
That's a fast reply! 
Looks like we are using active CF mappings which may be throwing it off(?).  I'm guessing I would need to add one for my API folder right? I'm relatively new to all this so my apologies.

**Comment by atuttle** [comment]
Well first things first, try getting everything working without any mappings/etc. (When I said symlinks, I did not mean mappings, btw. I meant symbolic links, which is a filesystem thing, not a CF thing.)

Put the taffy folder in your web root. Create an api folder in the web root, and in it put your Application.cfc and resources folder. If that works, then there's nothing wrong with your CF install/etc.

**Comment by codybe** [comment]
My mistake.  We are using no symlinks or aliases in the paths and as of yet I have not been able to get taffy to recognize the resources folder in my web root.

**Comment by atuttle** [comment]
to be clear, when I say you need to move stuff into the web root, here's what I mean:

```
/taffy/core/api.cfc
/taffy/...
/resources/...
/Application.cfc (extends=taffy.core.api)
/index.cfm
```

**Comment by codybe** [comment]
No luck with that either.  Still being told that there is no resources folder.

**Comment by atuttle** [comment]
In your Application.cfc, in onRequestStart, add this code:

`writeDump(var=guessResourcesFullPath(),abort=true);`

Does it guess correctly? If not, does it give you any clues about what might be wrong?

**Comment by codybe** [comment]
It is printing the correct path.

**Comment by atuttle** [comment]
Do your resource CFC's extend `taffy.core.resource`?

**Comment by codybe** [comment]
they do.

**Comment by timmixell** [comment]
Hey Adam,

I've been working with Taffy 3 for a new API, and everything was working swimmingly in local dev. Once we moved it up to testing, things started getting a bit hairy. In my local dev, I was getting to the Taffy app via directory: http://localhost/mytaffyapp/ - On testing, we've created a new site in IIS that points to the mytaffyapp directory: http://apittest.ourbiz.com/.

I'm using application-level mappings, so /taffy and /resources point to subdirectories within mytaffyapp. After doing some targeted dumping (heh), it looks like the issue is in the guessResourcesPath() method. Per the docs (http://docs.taffy.io/3.0.0/), this should be acceptable, but what I'm seeing when I do a dump within that method, is that it's truncating the complete path, omitting mytaffyapp altogether. So what should be c:/pathtowebroot/mytaffyapp/resources is returning as c:/pathtowebroot/resources. It _looks_ like Taffy isn't honoring the application-level mapping of /resources. If I hardcode guessResourcePath() to return "/resources" it works like a champ.

Obviously, I'd like to avoid customizing Taffy library code, I just wanted to make sure I wasn't doing anything wrong.

Thanks in advance!

**Comment by atuttle** [comment]
Thanks Tim, that's great debugging info. Unfortunately I'm in the middle of a time crunch for work so I don't have time to look at this. It sounds like what you've found is a valid concern, so feel free to try and work it out in a pull request if you would like to. It sounds like there needs to be a (better?) system for trying/verifying things before falling back to its current approach.

**Comment by timmixell** [comment]
Right on. I'll take a crack at it and submit it for your review.

Thanks a bunch!

**Comment by atuttle** [comment]
Does #268 resolve this issue?

**Comment by sashabass** [comment]
Guys,

I have had the same issue here as d1b1. I used to work quite extensively with Taffy 1.0 a while back and now came back to use the new 3.0 version. As fancy as I like to get when busting up a "Hello World" code in anything, I had made a mistake in my hello resource. It was a syntax mistake, I was trying to return a structure like this:

	public function get() {
		return representationOf({
			"name": "My name",
			"version": "1.0",
			"description": "The REST Web Service API"
		}).withStatus(200);
	}

So, obviously I confused JSON structure and CF structure formats. It should have been this:

	public function get() {
		return representationOf({
			"name" = "My name",
			"version" = "1.0",
			"description" = "The REST Web Service API"
		}).withStatus(200);
	}

For the longest time I couldn't figure out what was going on and I started debugging. Went down to taffy.core.factory.cfc, loadBeansFromPath method. As the code was looping through the resource folder, that it could comfortably recognize, it would go to cfcatch part of it after failing to create a resource bean, because if grammatical error I had in the code for hello.cfc resource.

Framework code in taffy.core.factory.cfc actually attempts to store the error message to:

application._taffy.status.skippedResources

object. However I didn't see it dumping application scope.

BOTTOM LINE: make sure you don't have stupid grammatical mistakes in your resources code, or else resources will not load.

Hopefully this helps.

**Comment by atuttle** [comment]
The skipped resources (and information on why they were skipped) are
displayed in the dashboard.

**Comment by atuttle** [comment]
Also, that colon-syntax is supported in more recent versions of ACF. I
think CF10+.

**Comment by sashabass** [comment]
I see, thanks for clarifications..
